doc/source/serve/deploy_model_tutorial.rst

```diff
@@ -0,0 +1,315 @@
+  "he planted the first human footprint on the lunar crust: \"That\'s one "   
+  "small step for man, one giant leap for mankind.\" His first step on the "  
+  "moon came at 10:56:20 P.M., as a television camera outside the craft "     
+  "transmitted his every move to an awed and excited audience of hundreds "   
+  "of millions of people on earth.")
+  
+  article_summary = summarize(article_text, 100)
+  
+  print(article_summary)
+
+  '''
+  Print output:
+
+  and walked on the moon. The astronauts of Apollo 11 steered their craft to
+  the historic landing yesterday at 4:17:40 P.M., Eastern daylight time.
+  "That's one small step for man, one giant leap for mankind," Armstrong
+  declared as he planted the first human footprint on the lunar crust. He
+  radioed to earth and the mission control room: "Houston, Tranquility Base
+  here. The Eagle has landed."
+  '''
+
+The ``tokenizer`` and ``model`` variables store our model's input encodings and weights.
+The ``summarize`` function takes in article ``text`` and the summary's ``max_length``, and then it summarizes the article text.
+In the example above, the ``summarize`` function generates a concise version of a lengthy news snippet about the moon landing.
+
+The goal is to deploy this model using Ray Serve, so it can be queried over HTTP.
+We'll start by converting the above Python function into a Ray Serve deployment that can be run locally on a laptop.
+
+First, we need to import ``ray``, ``ray serve``, and ``requests``.
+The ``ray`` and ``ray serve`` libraries give us access to Ray Serve's deployments, 
+so we can access our model over HTTP. The requests library handles HTTP request routing:
+
+.. code-block:: python
+
+  import ray
+  from ray import serve
+  import requests
+
+After these imports, we can include our model code from above. 
+We won't call our ``summarize`` function just yet though! 
+We will soon add logic to handle HTTP requests, so the ``summarize`` function 
+can operate on article text sent via HTTP request.
+
+.. code-block:: python
+
+  from transformers import AutoModelWithLMHead, AutoTokenizer
+
+  def summarize(text, max_length=150):
+    tokenizer = AutoTokenizer.from_pretrained("mrm8488/t5-base-finetuned-summarize-news")
+    ...
+    return preds[0]
+
+Ray Serve needs to run on top of a Ray Cluster, so we start a local one:
+
+.. code-block:: python
+
+  ray.init()
+
+.. note::
+
+  ``ray.init()`` will start a single-node Ray cluster on your local machine, 
+  which will allow you to use all your CPU cores to serve requests in parallel. 
+  To start a multi-node cluster, see :doc:`../cluster/index`.
+
+Next, we start the Ray Serve runtime:
+
+.. code-block:: python
+
+  serve.start()
+
+.. warning::
+
+  When the Python script exits, Ray Serve will shut down.  
+  If you would rather keep Ray Serve running in the background you can use 
+  ``serve.start(detached=True)`` (see :doc:`deployment` for details).
```

Comment:
We should probably use serve.start(detached=True) as the default in the code and then add a snippet about how users can also use serve.start() [without detached] since thats the primary use case